### PR TITLE
o365,sentinel_one_cloud_funnel,sysmon_linux,system,windows: tighten ipv4 extraction

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

Make sure that the prefix of the IP address is ::ffff in all cases.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Ref #10801

### CODE REVIEW

**Review by marc-gr — APPROVED** [review]

**Review by leehinman — COMMENTED** [review]
I'm ok with code as proposed but I'm wondering about the value of an improvement.  
Because the regex could match things like 999.999.999.999, maybe it would be a good idea to save the IP address to a temp field, and then use the convert processor to make sure it is a valid IP address.  We could make the regex tighter, but that gets harder and harder for the human to read

**Comment by efd6** [comment]
There are multiple things that are wrong with this approach, but they all boil down to attempting to parse IPs by RE. The intention here is not to filter incorrect IP values (e.g. a 999.999.999.999), but rather to ensure that correctly formatted IPs that are not IP4-mapped IPv6 addresses are not treated as though they are. The risk of that is low even without this change as noted in the original PR, but I thought that this was a reasonably straightforward addition that would prevent it.

**Review by leehinman — APPROVED** [review]

**Comment by andrewkroh** [comment]
/test benchmark fullreport

(wondering why sentinel_one_cloud_funnel wasn't included in the earlier output)

**Comment by efd6** [comment]
I am unable to replicate this failure locally.

**Review by chrisberkhout — COMMENTED** [review]
Should it accept enclosing square brackets without a port?
If so I'd suggest this:
```diff
-         pattern: '^\[?::ffff:([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)(?:\]:[0-9]+)?$'
+         pattern: '^\[?::ffff:([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)\]?(?::[0-9]+)?$'
```
Would be handy to see a list of example values for it to parse.

**Comment by efd6** [comment]
From a defensive programming perspective, I think that's a good idea. Though it should be `'^\[?::ffff:([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)(?:\](?::[0-9]+)?)?$'` to ensure that the port being present implies the bracket is present.

**Review by ishleenk17 — APPROVED** [review]

**Review by nfritts — APPROVED** [review]

File: packages/o365/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -915,9 +915,10 @@ processors:
           }
         }
   - gsub:
+      tag: extract_ipv4_and_port
       field: client._temp
-      pattern: '::ffff:([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)'
-      replacement: '$1'
+      pattern: '^\[?::ffff:([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)(?:\](:[0-9]+)?)?$'
+      replacement: '$1$2'
```
Comment (nicpenning) [inline packages/o365/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:921]: @efd6 - curious why you have $2 in this integration but only $1 in the rest. Care to elaborate for my understanding?

Comment (efd6) [reply packages/o365/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:921]: This one actually uses the port value, the rest only check that it's there.

Comment (nicpenning) [reply packages/o365/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:921]: Thank you.

**Pull request merged**